python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -111,6 +113,38 @@ def __new__(
         )
 
 
+@whitelist_for_serdes
+class NotABackfillRequest(
+    NamedTuple(
+        "_NotABackfillRequest",
+        [
+            ("asset_graph_subset", "AssetGraphSubset"),
+            ("tags", PublicAttr[Optional[Mapping[str, str]]]),
+            ("title", PublicAttr[Optional[str]]),
+            ("description", PublicAttr[Optional[str]]),
+        ],
+    )
+):
+    @property
+    def asset_keys(self) -> AbstractSet[AssetKey]:
+        return self.asset_graph_subset.asset_keys
+
+
+# @whitelist_for_serdes
+# @dagster_model
+# class NotABackfillRequest:
+#     asset_graph_subset: "AssetGraphSubset"
```

Comment:
dagster_model has an issue with type hints in "". If/when that gets fixed, then I'll switch to this version